base/compiler/init.jl

```diff
@@ -0,0 +1,268 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+using Core.Intrinsics
+import Core: print, println, show, write, unsafe_write, STDOUT, STDERR,
+             _apply, svec, apply_type, Builtin, IntrinsicFunction, MethodInstance
+
+ccall(:jl_set_istopmod, Cvoid, (Any, Bool), Compiler, false)
+
+eval(x) = Core.eval(Compiler, x)
+eval(m, x) = Core.eval(m, x)
+
+include(x) = Core.include(Compiler, x)
+include(mod, x) = Core.include(mod, x)
+
+inlining_enabled() = (JLOptions().can_inline == 1)
+coverage_enabled() = (JLOptions().code_coverage != 0)
+
+function return_type end
+
+#################################
+# essential files and libraries #
+#################################
+
+include("../essentials.jl")
+include("../ctypes.jl")
+include("../generator.jl")
+include("../reflection.jl")
+include("../options.jl")
+
+###########################
+# core operations & types #
+###########################
+
+include("../promotion.jl")
+include("../tuple.jl")
+include("../pair.jl")
+include("../traits.jl")
+include("../range.jl")
+include("../expr.jl")
+include("../error.jl")
+
+###################################
+# core numeric operations & types #
+###################################
+
+include("../bool.jl")
+include("../number.jl")
+include("../int.jl")
+include("../operators.jl")
+include("../pointer.jl")
+
+#########################
+# core array operations #
+#########################
+
+include("../indices.jl")
+include("../array.jl")
+include("../abstractarray.jl")
+
+########################
+# map-reduce operators #
+########################
+
+macro simd(forloop)
+    esc(forloop)
+end
+include("../reduce.jl")
+
+###################
+# core structures #
+###################
+
+include("../bitarray.jl")
+include("../bitset.jl")
+include("../abstractdict.jl")
+include("../iterators.jl")
+include("../namedtuple.jl")
+
+##################
+# core docsystem #
+##################
+
+include("../docs/core.jl")
+
+#############
+# constants #
+#############
+
+const EMPTY_VECTOR = Vector{Any}()
+
+const COMPILER_TEMP_SYM = Symbol("#temp#")
+
+const CoreNumType = Union{Int32, Int64, Float32, Float64}
+
+const isleaftype = _isleaftype
+
+const getproperty = getfield
+
+const setproperty! = setfield!
+
+const checked_add = +
+
+const checked_sub = -
+
+#####################
+# CompilationResult #
+#####################
+
+mutable struct CompilationResult
```

Comment:
The name of this should be reverted. It doesn't give you compiled code, just inferred code. Also only used by inference, so it can go in that file.